### PR TITLE
UNTRACKED/k1ch/refactor-persona-db-layer

### PR DESCRIPTION
**Pull request by k1ch**
# Overview
- [refactor: UNTRACKED/k1ch/refactor-persona-db-layer](https://github.com/DMGT-TECH/the-usher-server/commit/aaec4127a421a57a4042e08a9207a57ece27ab95)

## Your PR Checklist 🚨

❤️ Please review the [guidelines for contributing](../docs/CONTRIBUTING.md) to this repository. Our goal is to merge PRs fast 💨 .

- [x] Check your code additions will fail neither code linting checks nor unit tests.
- [ ] Additional units tests have been added to prove code updates work and fixes are effective
- [x] Additional documentation has been added (if appropriate)
- [x] Update Assignee field to yourself

## Issue Reference

Closes #

### CODE REVIEW

File: database/knexfile.js [hunk 1]
```diff
@@ -6,6 +6,10 @@ module.exports = {
   searchPath: [env.PGSCHEMA, 'public'],
   migrations: {
     tableName: 'knex_migrations',
-    schemaName: env.PGSCHEMA
-  }
+    schemaName: env.PGSCHEMA,
+  },
+  pool: {
+    min: 1,
+    max: 100,
```
Comment (mikelax) [inline database/knexfile.js:13]: @k1ch  are these the default values for knex? Should we make these load from environment variables, and then if not specified fallback to a default value of 1 and 100?

**Review by mikelax — APPROVED** [review]
@k1ch  was this change to address some perf issues we were seeing with using pg directly?

File: database/knexfile.js (hunk 1 above)
Comment (k1ch) [reply database/knexfile.js:13]: The default values are `min: 2` and `max: 10`. Good idea I will introduce ENV variable to easily change configuration.

**Comment by k1ch** [comment]
> @k1ch  was this change to address some perf issues we were seeing with using pg directly?

Yes. It's an effort to optimize server connections to the database. The `PGPool` is no longer used for `/personas` APIs.

**Comment by mikelax** [comment]
> > @k1ch  was this change to address some perf issues we were seeing with using pg directly?
> 
> Yes. It's an effort to optimize server connections to the database. The `PGPool` is no longer used for `/personas` APIs.

Cool, do you want to also refactor the way the db functions are handling / throwing errors or rather keep that for a separate PR?

**Comment by k1ch** [comment]
> > > @k1ch  was this change to address some perf issues we were seeing with using pg directly?
> > 
> > 
> > Yes. It's an effort to optimize server connections to the database. The `PGPool` is no longer used for `/personas` APIs.
> 
> Cool, do you want to also refactor the way the db functions are handling / throwing errors or rather keep that for a separate PR?

I prefer to keep that task for a separate PR since some of the test logic is based on these string values!

**Review by mikelax — APPROVED** [review]

**Pull request merged**